contract/services.py

```diff
@@ -1,41 +1,157 @@
+                datetime=core.datetime.datetime.now(),
+                message=""
+            ), cls=DjangoJSONEncoder)
+            c.save(username=self.user.username)
+            uuid_string = str(c.id)
+            # check if the PH is set
+            if "policy_holder_id" in contract:
+                # run services updateFromPHInsuree and Contract Valuation
+                total_amount = self.__services_update_from_ph_contract_valuation(
+                    policy_holder=contract["policy_holder_id"],
+                    contract_id=uuid_string
+                )
+                c.amount_notified = total_amount
+                c.save(username=self.user.username)
+            dict_representation = model_to_dict(c)
+            dict_representation['id'], dict_representation['uuid'] = (str(uuid_string), str(uuid_string))
+        except Exception as exc:
+            return _output_exception(model_name="Contract", method="create", exception=exc)
+        return _output_result_success(dict_representation=dict_representation)
+
+    def __services_update_from_ph_contract_valuation(self, policy_holder, contract_id):
+        cd = ContractDetails(user=self.user)
+        result_ph_insuree = cd.update_from_ph_insuree(contract_details={
+            "policy_holder_id": policy_holder,
+            "contract_id": contract_id,
+        })
+        ccpd = ContractContributionPlanDetails(user=self.user)
+        result_contract_valuation = ccpd.contract_valuation(
+            contract_contribution_plan_details={
+                "contract_details": result_ph_insuree["data"],
+                "save": False,
+            }
+        )
+        return result_contract_valuation["data"]["total_amount"]
 
     def submit(self, submit):
         pass
 
     def amend(self, submit):
         pass
 
-    def create(self, contract):
-        pass
-
     def update(self, contract):
         pass
 
     def delete(self, contract):
         pass
 
 
-@core.comparable
-class PolicyHolderContractDetails(object):
-
-    def __init__(self, policy_holder):
-        self.policy_holder = policy_holder
-        pass
-
-    def __eq__(self, other):
-        return isinstance(other, self.__class__) and self.__dict__ == other.__dict__
+class ContractDetails(object):
+
+    def __init__(self, user):
+        self.user = user
+
+    @check_authentication
+    def update_from_ph_insuree(self, contract_details):
+        try:
+            contract_insuree_list = []
+            policy_holder_insuree = PolicyHolderInsuree.objects.filter(policy_holder__id=contract_details['policy_holder_id'])
+            for phi in policy_holder_insuree:
+                cd = ContractDetailsModel(
+                    **{
+                        "contract_id": contract_details["contract_id"],
+                        "insuree_id": phi.insuree.id,
+                        "contribution_plan_bundle_id": str(phi.contribution_plan_bundle.id),
+                    }
```

Comment:
We need to get the calculation parameters that are saved in the PHinsuree json_ext

Reply:
Please at least add a # TODO

Reply:
ok.